spec.html

```diff
@@ -3712,75 +3712,93 @@ <h1>UpdateEmpty ( _completionRecord_, _value_ )</h1>
-      <p>A <dfn id="super-reference">Super Reference</dfn> is a Reference that is used to represent a name binding that was expressed using the super keyword. A Super Reference has an additional thisValue component, and its base value component will never be an Environment Record.</p>
-      <p>The following abstract operations are used in this specification to operate on references:</p>
+    <emu-clause id="sec-reference-record-specification-type" oldids="sec-reference-specification-type">
+      <h1>The Reference Record Specification Type</h1>
+      <p>The <dfn>Reference Record</dfn> type is used to explain the behaviour of such operators as `delete`, `typeof`, the assignment operators, the `super` keyword and other language features. For example, the left-hand operand of an assignment is expected to produce a Reference Record.</p>
+      <p>A Reference Record is a resolved name or property binding; its fields are defined by <emu-xref href="#table-reference-record-fields"></emu-xref>.</p>
 
-      <emu-clause id="sec-getbase" aoid="GetBase" oldids="ao-getbase">
-        <h1>GetBase ( _V_ )</h1>
-        <p>The abstract operation GetBase takes argument _V_. It performs the following steps when called:</p>
-        <emu-alg>
-          1. Assert: Type(_V_) is Reference.
-          1. Return the base value component of _V_.
-        </emu-alg>
-      </emu-clause>
-
-      <emu-clause id="sec-getreferencedname" aoid="GetReferencedName" oldids="ao-getreferencedname">
-        <h1>GetReferencedName ( _V_ )</h1>
-        <p>The abstract operation GetReferencedName takes argument _V_. It performs the following steps when called:</p>
-        <emu-alg>
-          1. Assert: Type(_V_) is Reference.
-          1. Return the referenced name component of _V_.
-        </emu-alg>
-      </emu-clause>
+      <emu-table id="table-reference-record-fields" caption="Reference Record Fields">
+        <table>
+          <tbody>
+            <tr>
+              <th>Field Name</th>
+              <th>Value</th>
+              <th>Meaning</th>
+            </tr>
+            <tr>
+              <td oldids="sec-getbase,ao-getbase">[[Base]]</td>
+              <td>
+                One of:
+                <ul>
+                  <li>
+                    any ECMAScript language value except *undefined* or *null*,
+                  </li>
+                  <li>
+                    an Environment Record, or
+                  </li>
+                  <li>
+                    ~empty~.
```

Comment:
I understand the desire to make `~empty~` a distinguished thing for many use cases, but in this case I feel like renaming this to `~unresolvable~` would be clearer.

Reply:
Sounds good to me; my only request was that it be a spec value and not `undefined`.

Reply:
Changed to `~unresolvable~` in 4ca96ec.